modules/imap/handler_modules.php

```diff
@@ -722,15 +722,23 @@ public function process() {
                 $this->session->set('imap_prefetched_ids', array_unique($prefetched, SORT_NUMERIC));
             }
             $with_subscription = isset($this->request->post['subscription_state']) && $this->request->post['subscription_state'];
+            $cache = Hm_IMAP_List::get_cache($this->cache, $form['imap_server_id']);
+            $imap = Hm_IMAP_List::connect($form['imap_server_id'], $cache);
+            if (imap_authed($imap)) {
+                $quota_root = $imap->get_quota_root('INBOX');
```

Comment:
It seems a folder can be posted from incoming params above - use it instead of defaulting to INBOX. Also, when you default to INBOX, why do you get quota again on first result returned? Seems wrong to me. I think if you have just one progress bar under all folders on the account, you should use root quota and not get individual folders' ones.